mantidproject/mantid
# BASISReduction: wrong EFixed for Si333 reflection

**Description of work.**
- [x] Fixed bug in source
- [x] System test for the 333 reflection

**To test:**
Donwload and unzip [golden.nxs.zip](https://www.dropbox.com/s/mxwvwszuizzjgfx/golden.nxs.zip?dl=0) into subdirectory `/tmp`. Then run the following script

```
golden=LoadNexus('/tmp/golden.nxs')
window=plotSpectrum(golden, 0)
BASISReduction(RunNumbers='90146',
               DoFluxNormalization=True,
               FluxNormalizationType='Monitor',
               ReflectionType='silicon_333',
               EnergyBins=[-330,0.4,330],
               MomentumTransferBins=[3.05, 4.30, 3.05])
plotSpectrum(mtd['BSS_90146_sqw'], 0, window=window)
```
and  verify both spectra overlap each other

Fixes #25588

This does not require release notes because it fixes a bug of the nightly version, not of the release version.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
